File: web-admin/lib/website/router/news.js
var router = require('koa-router')();
import { login_required } from '@server/common/middle/auth';

var website='/admin/website';

/**
 * 创建news
 */
router.post('/save-news',login_required,async(ctx,next)=>{
  const {id,title,source,source_url,pic,introduce,context,auth_data_code}=ctx.request.body;
  if(id){
    ctx.body=await ctx.postJson(website+'/update-news',{id,title,source,source_url,pic,introduce,context,auth_data_code});
  }else{
    ctx.body=await ctx.postJson(website+'/create-news',{title,source,source_url,pic,introduce,context,auth_data_code});
  }
});

/**
 * 获取详情 
 */
router.post('/get-news',login_required,async(ctx,next)=>{
  const {id}=ctx.request.body;
  ctx.body=await ctx.postString(website+`/get-news/${id}`,{});
});

/**
 * news删除 
 */
router.post('/del-news',login_required,async(ctx,next)=>{
  const {id}=ctx.request.body;
  ctx.body=await ctx.postString(website+`/del-news/${id}`,{});
});

router.post('/update-news-publish',login_required,async(ctx,next)=>{
  const {id,publish}=ctx.request.body;
  ctx.body=await ctx.postJson(website+`/update-news-publish`,{id,publish});
});

/**
 * news列表
 */
router.post('/list-news',login_required,async(ctx,next)=>{
  const {page_num,page_size}=ctx.request.body;
  ctx.body=await ctx.postJson(website+'/list-news',{page_num,page_size});
});



module.exports = router;

File: server/pro-api/src/main/java/com/pro/website/dto/entity/News.java
package com.pro.website.dto.entity;

import lombok.Getter;
import lombok.Setter;
import java.io.Serializable;
import java.time.LocalDateTime;

/**
 * @auth guozhenhua
 * @date 2019/07/12
 */
@Getter
@Setter
public class News  implements Serializable {


  /**
   * id号
   */

  private Long id;

  /**
   * 标题
   */

  private String title;

  /**
   * 是否发布
   */

  private Short publish;

  /**
   * 来源
   */
  private String source;

  /**
   * 来源地址
   */

  private String source_url;

  /**
   * 列表预览图片
   */

  private String pic;

  /**
   * 描述
   */

  private String introduce;

  /**
   * 内容
   */

  private String context;

  /**
   * 创建时间
   */

  private LocalDateTime create_time;

  /**
   * 更新时间
   */

  private LocalDateTime update_time;


  /**
   * 更新时间
   */
  private LocalDateTime publish_time;

  /**
   * 是否删除
   */

  private Short del;

  /**
   * 数据权限码
   */
  private String auth_data_code;

}
